classtest.js

```diff
@@ -1,87 +1,152 @@
+  this.hourlySales = [];
+}
+CookieStore.prototype.getAvgCookieCount = function (){
+  var range = this.maxCustomers - this.minCustomers;
+  return Math.floor(Math.random() * range + this.minCustomers);
+};
+CookieStore.prototype.renderRow = function() {
+  var rowEl = document.createElement('tr');
   var nameEl = document.createElement('th');
-  nameEl.textContent = currentStore.name;
+  nameEl.textContent = this.name + ':';
   rowEl.appendChild(nameEl);
 
-
+//--for loop to produce cph stats for each time slot for this row (it is inside the store loop so it will go 16 times five different times....)
   for(var j = 0; j < timeSlot.times.length ; j++){
+    var storeRandomCPH = this.getAvgCookieCount();
     var timeEl = document.createElement('td');
-    timeEl.textContent = cph();
+    timeEl.textContent = storeRandomCPH;
     rowEl.appendChild(timeEl);
-    total += cph();
+    total += storeRandomCPH;
+    this.hourlySales.push(storeRandomCPH);
+    console.log(this.hourlySales);
   }
-  //log to dom the total in a column
-  console.log(currentStore.name + ' ' + total);
+  //---render to dom the total in a column---//
   tableEl.appendChild(rowEl);
   var totalEl = document.createElement('td');
   totalEl.textContent = total;
   rowEl.appendChild(totalEl);
   total = 0;
 };
+var colTotalAdded = document.createElement('tr');
+colTotalAdded.textContent = colTotals;
+tableEl.appendChild(colTotalAdded);
 
-document.body.appendChild(tableEl);
+function columnTotals(){
+  for (var ii = 0; ii < timeSlot.times.length; ii++){
+    var allStoreHourlyTotal = 0;
+    for(var jj = 0; jj < stores.length; jj++){
+      allStoreHourlyTotal += stores[jj].hourlySales[ii];
+    }
+    colTotals.push(allStoreHourlyTotal);
+  };
+  console.log(colTotals);
+}
 
-//------------------------FUNCTIONS----------------------------
-function cph(){
-  var avgCookiesPerHour = Math.ceil((Math.floor(Math.random() * ( currentStore.maxCustomers - currentStore.minCustomers + 1) + currentStore.minCustomers)) * currentStore.avgCookies);
-  return avgCookiesPerHour;
-};
+console.log('------- EVENT LISTENERS--------');
+//---linking the js to the form tag and id on html---//
+var storeFormEl = document.getElementById('new-store-form');
+//---adding an event listener that when submit is executed it runs "handleSubmit" function
+storeFormEl.addEventListener('submit', handleSubmit);
+//---...---//
+function handleSubmit(event){
+  event.preventDefault();//---what does this do??---//
+  event.stopPropagation();//---what does this do??---//
+  var storeName = event.target.cookieStoreName.value;
+  var minCust = parseInt(event.target.minCustomersPerHour.value);
+  var maxCust = parseInt(event.target.maxCustomersPerHour.value);
+  var avgCookies = parseFloat(event.target.avgCookiesPerCustomer.value);
+  // console.log('User Pressed Submit Button on Form');
+  var store = new CookieStore(storeName, minCust, maxCust, avgCookies);
+  stores.push(store);
+  store.renderRow();
```

Comment:
In the long term, you may need to refactor this to use the drawTable function instead.